src/vs/platform/contextview/browser/contextViewService.ts

```diff
@@ -3,18 +3,17 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { ContextView, ContextViewDOMPosition } from 'vs/base/browser/ui/contextview/contextview';
+import { ContextView, ContextViewDOMPosition, IContextViewProvider } from 'vs/base/browser/ui/contextview/contextview';
 import { Disposable, IDisposable, toDisposable } from 'vs/base/common/lifecycle';
 import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
 import { IContextViewDelegate, IContextViewService } from './contextView';
 import { getWindow } from 'vs/base/browser/dom';
 
-export class ContextViewService extends Disposable implements IContextViewService {
 
-	declare readonly _serviceBrand: undefined;
+export class ContextViewHandler extends Disposable implements IContextViewProvider {
```

Comment:
Is there a way to remove the DOM elements when they are hidden? I think now we pile up context view DOM elements on the root for each instance.

Reply:
Not sure this is worth it for hover views as they can appear / reappear a lot. But makes sense for context views